pkg/auth/model/model.go

```diff
@@ -353,6 +344,16 @@ func ConvertCredList(creds []*DBCredential) []*Credential {
 	return res
 }
 
+func ConvertCredDataList(creds []proto.Message) []*Credential {
+	res := make([]*Credential, 0, len(creds))
+	for _, c := range creds {
+		a := c.(*CredentialData)
```

Comment:
Did you check if we can omit the SecretAccessKey from CredentialData and leave only the encrypted value?